eis_toolkit/prediction/weights_of_evidence.py

```diff
@@ -0,0 +1,255 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from beartype import beartype
+from beartype.typing import List, Literal, Optional, Sequence, Tuple
+
+from eis_toolkit import exceptions
+from eis_toolkit.vector_processing.rasterize_vector import rasterize_vector
+
+
+def _read_and_preprocess_evidence(raster: rasterio.io.DatasetReader, nodata: Optional[Number] = None) -> np.ndarray:
+    """Read raster data and handle NoData values."""
+
+    array = np.array(raster.read(1), dtype=np.float32)
+
+    if nodata is not None:
+        array[array == nodata] = np.nan
+    elif raster.meta["nodata"] is not None:
+        array[array == raster.meta["nodata"]] = np.nan
+
+    return array
+
+
+def _calculate_metrics_for_class(deposits: np.ndarray, evidence: np.ndarray) -> tuple:
```

Comment:
```suggestion
def _calculate_metrics_for_class(deposits: np.ndarray, evidence: np.ndarray) -> Tuple[float, float, float, float, float, float, float, float, float, float, float]:
```

Length of the tuple is known looks like. Therefore could explicitly state it.